src/CoreWCF.WebHttp/src/CoreWCF/Dispatcher/DataContractJsonSerializerOperationFormatter.cs

```diff
@@ -0,0 +1,818 @@
+    {
+        private readonly bool _isBareMessageContractReply;
+        private readonly bool _isBareMessageContractRequest;
+        // isWrapped is true when the user has explicitly chosen the response or request format to be Wrapped (allowed only in WebHttpBehavior)
+        private readonly bool _isWrapped;
+        private readonly bool _useAspNetAjaxJson;
+        //private readonly string _callbackParameterName;
+
+        private static readonly DataContractFormatAttribute s_defaultDataContractFormatAttribute = new DataContractFormatAttribute();
+
+        public DataContractJsonSerializerOperationFormatter(OperationDescription description, int maxItemsInObjectGraph, bool ignoreExtensionDataObject, bool isWrapped, bool useAspNetAjaxJson, string callbackParameterName)
+            : base(description, s_defaultDataContractFormatAttribute, new DataContractJsonSerializerOperationBehavior(description, maxItemsInObjectGraph, ignoreExtensionDataObject, useAspNetAjaxJson ? EmitTypeInformation.Always : EmitTypeInformation.Never))
+        {
+            if (requestMessageInfo != null)
+            {
+                if (requestMessageInfo.WrapperName == null)
+                {
+                    _isBareMessageContractRequest = true;
+                }
+                else
+                {
+                    requestMessageInfo.WrapperName = JsonGlobals.RootDictionaryString;
+                    requestMessageInfo.WrapperNamespace = XmlDictionaryString.Empty;
+                }
+            }
+
+            if (replyMessageInfo != null)
+            {
+                if (replyMessageInfo.WrapperName == null)
+                {
+                    _isBareMessageContractReply = true;
+                }
+                else
+                {
+                    if (useAspNetAjaxJson)
+                    {
+                        replyMessageInfo.WrapperName = JsonGlobals.DDictionaryString;
+                    }
+                    else
+                    {
+                        replyMessageInfo.WrapperName = JsonGlobals.RootDictionaryString;
+                    }
+                    replyMessageInfo.WrapperNamespace = XmlDictionaryString.Empty;
+                }
+            }
+
+            if ((requestStreamFormatter != null) && (requestStreamFormatter.WrapperName != null))
+            {
+                requestStreamFormatter.WrapperName = JsonGlobals.RootString;
+                requestStreamFormatter.WrapperNamespace = string.Empty;
+            }
+
+            if ((replyStreamFormatter != null) && (replyStreamFormatter.WrapperName != null))
+            {
+                replyStreamFormatter.WrapperName = JsonGlobals.RootString;
+                replyStreamFormatter.WrapperNamespace = string.Empty;
+            }
+
+            _isWrapped = isWrapped;
+            _useAspNetAjaxJson = useAspNetAjaxJson;
+            //_callbackParameterName = callbackParameterName;
```

Comment:
If looks like this is always passed in as false so can be removed and all the conditional code simplified to remove the true case.

Reply:
This is addressed in 2fe28110cbbcc5f9c7dfb3ff3b068cc829a4941b.